prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/FhirPathUtils.kt

```diff
@@ -145,7 +144,6 @@ object FhirPathUtils : Logging {
                 throw SchemaException("FHIR Path expression did not evaluate to a boolean type: $expression")
             }
         } catch (e: Exception) {
-            // This is due to a bug in at least the extension() function
```

Comment:
I think it's okay to leave as is but `is SchemaException -> throw e` as part of the when statement seems particularly odd to me.

Reply:
I agree, Brick. It looks like we throw a schemaException anyways (line 157) but if we actually get a schema exception (line 151) we throw early and hence do not hit the log code (line 156). 

proposed change: on line 151, change `throw e` to `e.msg`?

I think the reason we are doing the error handling this way with custom messages is we do not want to accidentally leak PII that would be contained in a FHIRLexerException (line 150) or other exceptions (line 152) but we know SchemaException won't contain PII so that's why it doesn't have a custom error message here.

Let me know your thoughts Josh, maybe you have a different understanding than me!

Reply:
Yeah I think this change makes sense and makes that part fit more with the pattern established in this catch (https://github.com/CDCgov/prime-reportstream/pull/15020/commits/8806331173c72e2f3ad86bef7f2bc2a82e12fb78). I made the change and will bring up leaking PII in stand up tomorrow. Seems like something we should document or tag better if it's possible in areas like this.